lemonspotter/samplers/valid.py

```diff
@@ -109,10 +109,15 @@ def generate_sample(self, parameter: Parameter) -> Iterable[Variable]:
         type_samples = []
 
         if parameter.direction == Direction.OUT:
-            # generate out variable
-            var = Variable(parameter.type, parameter.name + '_out')
+            if parameter.type.dereferencable:
```

Comment:
Thinking about it, we should generalise this to a factory function for variables. Let's do it like this and add a ticket to make a variable factory which emits as required variables.